charts/loki-distributed/templates/_helpers.tpl

```diff
@@ -141,6 +141,17 @@ Return the appropriate apiVersion for PodDisruptionBudget.
   {{- end -}}
 {{- end -}}
 
+{{/*
+Return the appropriate apiVersion for HorizontalPodAutoscaler.
+*/}}
+{{- define "loki.hpa.apiVersion" -}}
+  {{- if and (.Capabilities.APIVersions.Has "autoscaling/v2") (semverCompare ">=1.25-0" .Capabilities.KubeVersion.Version) -}}
```

Comment:
Why don't we just query api versions has `autoscaling/v2/HorizontalPodAutoscaler` ?

```suggestion
  {{- if .Capabilities.APIVersions.Has "autoscaling/v2/HorizontalPodAutoscaler -}}
```

Proof:

```
$ kind create cluster
Creating cluster "kind" ...
 ✓ Ensuring node image (kindest/node:v1.25.0) 🖼
 ✓ Preparing nodes 📦
 ✓ Writing configuration 📜
 ✓ Starting control-plane 🕹️
 ✓ Installing CNI 🔌
 ✓ Installing StorageClass 💾
Set kubectl context to "kind-kind"
You can now use your cluster with:

kubectl cluster-info --context kind-kind

Not sure what to do next? 😅  Check out https://kind.sigs.k8s.io/docs/user/quick-start/
$ k api-resources | grep autoscaling/v2
horizontalpodautoscalers          hpa          autoscaling/v2                         true         HorizontalPodAutoscaler
```

Reply:
Yes, it can be done this way too, but I wanted to use the same syntax as in the already existing define apiVersion code blocks. I can rework it to the desired state, but is this really the wanted change @zanhsieh? Or I can refactor each occurrence to the desired state to preserve consistency.

Reply:
I really only want to make possible to use HPA on the k8s >=1.25 without huge changes and with same functionality. Please can you create enhancement issue for your request? I will try to focused on it later and add also another new features as custom and multiple metrics.